Enable Override of EBS Region, AZ, & Endpoint Props

akutz opened the pull request:
This patch introduces the ability for the EBS driver to override the client's provided region, availability zone, and endpoint properties by specifying them in the server-side config with a suffix of !important (like CSS).

Many thanks to libStorage & REX-Ray community member @shamimgeek for discovering this! Thanks Shamim!

codenrhoden reviewed (commented):
Looks good at first glance. Definitely want to hear it works from @shamimgeek. This will be a nice way to pick up eucalyptus support!

akutz committed:
Enable Override of EBS Region, AZ, & Endpoint Props

This patch introduces the ability for the EBS driver to override the
client's provide region, availability zone, and endpoint properties by
specifying them in the server-side config with a suffix of !important
(like CSS).
